### PR TITLE
Inline images mail slow to display

### PR DESCRIPTION
**Pull request by adrien-coye**
## Abstract 
Performance issues on displaying large emails with inline pictures.
The mainthread was pinned, batching image processing greatly improves the UX in my testing.

## Misc
Mutating the dom of the email is super costly, therefore I used batching to process images and mutate less the content of the email view that is rendered on the main thread.
I went from 300 sec to 50 ish for a mail with a lot of small images, while keeping the UI responsive

### CODE REVIEW

File: Mail/Views/Thread/MessageView.swift [hunk 1]
```diff
@@ -100,22 +134,52 @@ struct MessageView: View {
...
+                        of: "cid:\(contentId)",
+                        with: "data:\(attachment.mimeType);base64,\(data.base64EncodedString())"
+                    )
+                }
+
+                // Mutate DOM
+                self.insertInlineAttachment(body: body, compactBody: compactBody)
+                
+                // Delay between each chunk processing just enough, so the user feels the UI is responsive.
+                try await Task.sleep(nanoseconds: 4_000_000_000)
```
Comment (adrien-coye) [inline Mail/Views/Thread/MessageView.swift:173]: Nice to have, use a debounce instead maybe ?

File: Mail/Views/Thread/MessageView.swift [hunk 2]
```diff
@@ -25,6 +25,40 @@ import RealmSwift
 import Shimmer
 import SwiftUI
 
+// TODO: move to Core
```
Comment (adrien-coye) [inline Mail/Views/Thread/MessageView.swift:28]: TODO: move to Core

Comment (adrien-coye) [reply Mail/Views/Thread/MessageView.swift:28]: Moved to core, but kMail is locked on an old ios-Core, so we keep it here in the meantime.

File: Mail/Views/Thread/MessageView.swift [hunk 3]
```diff
@@ -100,22 +134,52 @@ struct MessageView: View {
...
-
-                presentableBody.body?.value = presentableBody.body?.value?.replacingOccurrences(
-                    of: "cid:\(contentId)",
-                    with: "data:\(attachment.mimeType);base64,\(attachmentData.base64EncodedString())"
-                )
-                presentableBody.compactBody = presentableBody.compactBody?.replacingOccurrences(
-                    of: "cid:\(contentId)",
-                    with: "data:\(attachment.mimeType);base64,\(attachmentData.base64EncodedString())"
-                )
+        Task {
```
Comment (adrien-coye) [inline Mail/Views/Thread/MessageView.swift:137]: try detached

**Review by Ambrdctr — APPROVED** [review]

**Pull request merged**